x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/enrollment/CreateEnrollmentTokenAction.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.enrollment;
+
+import org.elasticsearch.action.ActionType;
+
+/**
+ * ActionType for creating an enrollment new token
+ */
+public class CreateEnrollmentTokenAction extends ActionType<CreateEnrollmentTokenResponse> {
+    public static final String NAME = "cluster:admin/xpack/security/enrollment/create";
```

Comment:
Let's discuss where these should live. Do you have any arguments for the enrollment actions being in the security plugin instead of core ? I added the APIs I implemented in core as it made more sense to me since they are cluster APIs but happy to change my mind

Reply:
Yang in https://github.com/elastic/elasticsearch/pull/72129#discussion_r620786962 and Tim in a followup changed my mind. Let's move this back to security, apologies for the back and forth :)

Reply:
See the discussions in https://github.com/elastic/elasticsearch/pull/72129#discussion_r627099480 about the action name